lib/client/ws_client.rb

```diff
@@ -9,50 +9,34 @@
 module ScaleRb
   class WsClient
     def self.start(url)
-      Async do |task|
+
+      Sync do |task|
         endpoint = Async::HTTP::Endpoint.parse(url, alpn_protocols: Async::HTTP::Protocol::HTTP11.names)
-        client = WsClient.new
-
-        task.async do
-          Async::WebSocket::Client.connect(endpoint) do |connection|
-            Async do
-              while request = client.next_request
-                ScaleRb.logger.debug "Sending request: #{request.to_json}"
-                connection.write(request.to_json)
-              end
-            end
 
-            # inside main task
+        Async::WebSocket::Client.connect(endpoint) do |connection|
+          client = WsClient.new(connection)
+
+          recv_task = task.async do
             while message = connection.read
-              data = JSON.parse(message)
+              data = message.parse
               ScaleRb.logger.debug "Received message: #{data}"
 
-              Async do
+              task.async do
                 client.handle_response(data)
```

Comment:
You'd want to add error handling here - i.e. what happens if `handle_response(data)` fails?

Reply:
yes, I'll add error handling here